src/Web/E-Games.Web/E-Games.Web/Controllers/UserController.cs

```diff
@@ -0,0 +1,108 @@
+﻿using E_Games.Data.Data.Models;
+using E_Games.Web.ViewModels;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using System.Security.Claims;
+
+namespace E_Games.Web.Controllers
+{
+    [ApiController]
+    [Route("api/user")]
+    public class UserController : ControllerBase
+    {
+        private readonly UserManager<ApplicationUser> _userManager;
+
+        public UserController(UserManager<ApplicationUser> userManager)
+        {
+            _userManager = userManager;
+        }
+
+        [HttpGet]
+        [Authorize]
+        public async Task<IActionResult> UserProfile()
+        {
+            var userId = User.FindFirstValue(ClaimTypes.NameIdentifier);
+            var user = await _userManager.FindByIdAsync(userId!);
+
+            if (user == null)
+            {
+                return NotFound(new { Message = "User not found" });
+            }
+
+            var userProfile = new UserProfileModel()
```

Comment:
use automapper fot this conversions. This way, it will be much easier to read code and add new properties